core/Archive.php

```diff
@@ -196,7 +197,7 @@ protected function __construct(Parameters $params, $forceIndexedBySite = false,
      *                             or date range (ie, 'YYYY-MM-DD,YYYY-MM-DD').
      * @param bool|false|string $segment Segment definition or false if no segment should be used. {@link Piwik\Segment}
      * @param bool|false|string $_restrictSitesToLogin Used only when running as a scheduled task.
-     * @return Archive
+     * @return static
```

Comment:
Is there a reason for this other than for tests? If so, it might be good to explain the reasoning/use-cases for deriving from Archive in the PHP docs for the class.

If not, it might be cleaner to abstract the ::build/::factory methods to a new ArchiveFactory class that can be overridden via DI.